.github/workflows/tests_and_cov.yml

```diff
@@ -1,47 +1,93 @@
-name: Run tests and upload coverage
+name: Run tests and push coverage to Codecov
 
-on: ["push", "pull_request"]
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
 
 jobs:
-  build:
-    name: Coverage
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install deps
+        run: |
+          pip install --upgrade --upgrade-strategy eager -r requirements-dev.txt -e .
+          pip check
+
+        # Cache package installation step to speed up the following step
+      - uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements-dev.txt') }}
+
+  test:
+    needs: setup
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
```

Comment:
Whats the default now? 3.10?

Reply:
3.9 perhaps? https://packages.ubuntu.com/impish/python3